pl-backend/Services/UserService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using pl_backend.Data;
+using pl_backend.DTO;
+using pl_backend.Models;
+using System.Data;
+using Microsoft.EntityFrameworkCore;
+using System.Security.Cryptography;
+
+namespace pl_backend.Services
+{
+    public interface IUserService
+    {
+        Task<User> Authorization(UserDto userDto);
+        Task<string> Login(UserDto userDto);
+        Task<User> GetUserId(int Id);
+    }
+    public class UserService : IUserService
+    {
+        private readonly DataContext _dataContext;
+        private readonly ITokenService _tokenService;
+        public UserService(DataContext dataContext, ITokenService tokenService)
+        {
+            _dataContext = dataContext;
+            _tokenService = tokenService;
+        }
+
+        public async Task<User> Authorization(UserDto userDto)
+        {
+            User user = new User();
+
+            if (userDto.Password.Length < 8) throw new Exception("Password is too short! It must be 8 characters long!");
+
+            CreatePasswordHash(userDto.Password, out byte[] passwordHash, out byte[] passwordSalt);
+
+            user.FirstName = userDto.FirstName;  //Potrzebujemy Id czy FirstName??????
+            user.PasswordHash = passwordHash;
+            user.PasswordSalt = passwordSalt;
```

Comment:
Id will be auto-generated when new you record is placed in db
```suggestion
            User user = new()
            {
                Email = userDto.FirstName,
                PasswordHash = passwordHash,
                PasswordSalt = passwordSalt
            };
```

Reply:
can you elaborate on this?